applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -693,28 +619,26 @@ static int update_sink_stream_qos(struct le_audio_headset *headset, uint32_t pre
 static void unicast_client_location_cb(struct bt_conn *conn, enum bt_audio_dir dir,
 				       enum bt_audio_location loc)
 {
-	int index = discover_list_index_get(conn);
+	int ret;
+	uint8_t index;
+
+	ret = channel_index_get(conn, &index);
 
-	if (index < 0) {
-		LOG_WRN("Failed to get discover_list index");
+	if (ret) {
+		LOG_ERR("Channel index not found");
 		return;
 	}
 
-	if ((loc & BT_AUDIO_LOCATION_FRONT_LEFT) || (loc & BT_AUDIO_LOCATION_SIDE_LEFT)) {
-		if (headsets[AUDIO_CH_L].headset_conn == NULL) {
-			headsets[AUDIO_CH_L].headset_conn = conn;
-			headsets[AUDIO_CH_L].waiting_for_sink_disc = discover_list[index].sink;
-			headsets[AUDIO_CH_L].waiting_for_source_disc = discover_list[index].source;
-		}
+	if ((loc & BT_AUDIO_LOCATION_FRONT_LEFT) || (loc & BT_AUDIO_LOCATION_SIDE_LEFT) ||
+	    (loc == BT_AUDIO_LOCATION_MONO_AUDIO)) {
+		headsets[index].location = BT_AUDIO_LOCATION_FRONT_LEFT;
+		headsets[index].ch_name = "LEFT";
 
 	} else if ((loc & BT_AUDIO_LOCATION_FRONT_RIGHT) || (loc & BT_AUDIO_LOCATION_SIDE_RIGHT)) {
-		if (headsets[AUDIO_CH_R].headset_conn == NULL) {
-			headsets[AUDIO_CH_R].headset_conn = conn;
-			headsets[AUDIO_CH_R].waiting_for_sink_disc = discover_list[index].sink;
-			headsets[AUDIO_CH_R].waiting_for_source_disc = discover_list[index].source;
-		}
+		headsets[index].location = BT_AUDIO_LOCATION_FRONT_RIGHT;
```

Comment:
Could you elaborate on the "overwrite" of headsets[index].location with BT_AUDIO_LOCATION_FRONT_RIGHT if it was BT_AUDIO_LOCATION_SIDE_RIGHT? Perhaps we can do this elsewhere and/or assign other locations as well to the "default" ones?